cli/internal/packager/seed.go

```diff
@@ -124,89 +47,26 @@ func preSeedRegistry(tempPath tempPaths) {
-		inject.args = []string{"ctr", "images", "import", tempPath.seedImages}
+	if clusterArch != state.Architecture {
+		message.Fatalf(nil, "The current Zarf package architecture %s does not match the cluster architecture %s", state.Architecture, clusterArch)
+	}
 
-	case k8s.DistroIsK3d:
+	switch state.Distro {
+	case k8s.DistroIsK3s, k8s.DistroIsK3d:
 		state.StorageClass = "local-path"
-		clusterName := getClusterName("k3d")
-		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
-		inject.command = "k3d"
-		inject.args = []string{"images", "import", tempPath.seedImages, "--cluster", clusterName}
 
-	case k8s.DistroIsKind:
+	case k8s.DistroIsKind, k8s.DistroIsGKE:
 		state.StorageClass = "standard"
-		// See https://github.com/kubernetes-sigs/kind/blob/v0.11.1/pkg/cluster/internal/kubeconfig/internal/kubeconfig/helpers.go#L24
-		clusterName := getClusterName("kind")
-		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
-		inject.command = "kind"
-		inject.args = []string{"load", "image-archive", tempPath.seedImages, "--name", clusterName}
 
 	case k8s.DistroIsDockerDesktop:
 		state.StorageClass = "hostpath"
-		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
-		inject.command = "docker"
-		inject.args = []string{"load", "-i", tempPath.seedImages}
 
-	case k8s.DistroIsMicroK8s:
-		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
-		inject.command = "microk8s"
-		inject.args = []string{"ctr", "images", "import", tempPath.seedImages}
-
-	default:
-		state.Registry.SeedType = config.ZarfSeedTypeRuntimeRegistry
 	}
 
-	switch state.Registry.SeedType {
-	case config.ZarfSeedTypeCLIInject:
-		var (
-			output  string
-			spinner = message.NewProgressSpinner("Injecting Zarf registry image using %s", inject.command)
-		)
-		defer spinner.Stop()
-
-		// If this is a seed image injection, attempt to run it and warn if there is an error
-		output, err = utils.ExecCommand(false, nil, inject.command, inject.args...)
-		message.Debug(output)
-		if err != nil {
-			spinner.Errorf(err, "Unable to inject the seed image from the %s archive", tempPath.seedImages)
-			spinner.Stop()
-		} else {
-			spinner.Success()
-		}
-
-		// Set TLS host so that the seed template isn't broken
-		config.TLS.Host = config.IPV4Localhost
-
-	case config.ZarfSeedTypeRuntimeRegistry:
-		// Otherwise, start embedded registry read/write (only on localhost)
-		startSeedRegistry(config.IPV4Localhost, false)
-
-		// Populate the seed registry
-		images.PushToZarfRegistry(tempPath.seedImages, config.GetSeedImages(), config.ZarfLocalSeedRegistry)
-
-		// Close this registry now
-		stopSeedRegistry()
-
-		if config.TLS.Host == "" {
-			// Get user to choose/enter host info for the read-only seed registry
-			tls.HandleTLSOptions(config.DeployOptions.Confirm)
-			pki.HandlePKI()
-		}
-
-		// Start the registry again read-only now
-		startSeedRegistry(config.TLS.Host, true)
-
-	default:
-		message.Fatalf(nil, "Unknown seed registry status")
-	}
+	runInjectionMadness(tempPath)
```

Comment:
This is really sensitive when initializing with the k3s component. It seems like sometimes it is trying to create an injector pod when there aren't an existing images in the cluster yet and is stopping with a generic `Unable to perform the injection` fatal error. When adding a small sleep statement before this function call I haven't had any issues.

Reply:
this now waits for images to exist